### DISCUSSION TITLE
windows - use git bash as shell  via set shell

### DISCUSSION THREAD

**Discussion by bjoernbuth** [gokcehan/lf#1425] [Q&A] [answered]
Hi, thx for making this great tool to the devs and all that helped. 

I'm using it on **windows 10** and would like to be able to use it inside **git bash**. Is there any chance to do so ? 

I tried by setting the shell command in the lfrc file file in ...\AppData\Local\lf\lfrc  via 
- set shell "c/Program Files/Git/bin/sh.exe"  
- set shell "c:/Program Files/Git/bin/sh.exe"
and many many variations of this (c:/   , using backslashes  , different kinds of quoting, different locations,symlinks and so on)  but never got it to work.

I either get errors of 3 types when pressing w (which should run a shell): 
1. ...sh.exe does not exist 
1.  running shell .... exit status 126, this occure for exampe when trying (set shell bash.exe). 
1.  running shell .... exit status 1 (for set shell "c:/bash.exe" , which is admittedly a pretty desperate try considering I simply copied bash.exe there)

I would love to be able to use lf in git bash and thankful for help.

**Comment by joelim-work** [comment, marked as answer]
On Windows the `shellflag` option defaults to `/c`, have you tried changing it?

```
set shell 'C:\Program Files\Git\bin\sh.exe'
set shellflag '-c'
```

**Reply by bjoernbuth** [reply]
Thank you,  this is working. Using this I can e.g. do $bash (I've mapped it to w as this seems a common shortcut and makes it easier to run bash) and inside the bash shell that shows up I have access to $f, $fx etc. which gives me many possibilities for further customization. Thanks again.
